### DISCUSSION TITLE
Access "deeply nested" property

### DISCUSSION THREAD

**Discussion by alexandremjacques** [quasarframework/quasar#10509] [General - Components / Directives / etc] [answered]
Hi, I have a very simple page that I hoped was quite trivial. I fecth an JSON object from an API with a structure:
```
{
    "id: "123123",
    "customer": {
        "name": "Person Name"
    },
   "agency": {
   "legalName: "Legal name"
    },
    "purchaseDate": "2021-09-19"
    ...
} 
```
I get that on my `setup()` method:
```
  setup (props) {
    const state = reactive({
      sale: {}
    })

    onMounted(async () => {
      state.sale = await salesApi.getSaleDetails(props.saleId)
    })

    return {
      state
    }
```

My problem is: whener I try to access the customer name from the template using `{{ state.sale.customer.name }}`, I get an exception saying that `$setup.state.sale.customer is undefined`
If I stop at the customer object like `{{ state.sale.customer }}` I get the expected JSON part (with the name property inside it as it should).

Thanks in advance.

Regards!

**Comment by pikil** [comment, marked as answer]
@alexandremjacques The thing is that the template is accessing these props right at the moment mounted is triggered, meaning that `state` should already contain these keys inside (from what I see, it contains nothing - { sale: {} })

Try to add
```
{
 sale: {
  customer: {
   name: '',
   address: '',
   ...
  }
 }
}
```

inside and check if that helps?

**Reply by alexandremjacques** [reply]
Hi @pikil. I don't think that's the case since I can reach `state.sale` and `state.sale.customer`. Just as I try to access any field of the `customer` object, I get that error.

**Reply by pikil** [reply]
Yes, there is no problem of reaching ```customer``` property, as it is ```undefined```, and that is not breaking the convention. However, you are trying to access the non existent property of ```customer``` (undefined and null cannot have properties) and that is where the problem might be, from what I feel.

Can you please create codepen or jsfiddle, so we can look into that _deep nesting_ as well?
* You can use settimeout to simulate the XHR request :)
